File: Binary-Tree-BFS/binary-tree-level-order-traversal.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(!root) return {};
        std::queue<TreeNode*> q = {};
        q.push(root);
        vector<vector<int>> res = {};

        while(!q.empty())
        {
            size_t level_size = q.size();
            vector<int> temp = {};

            for(size_t i = 0; i < level_size; ++i)
            {
                TreeNode* popped = q.front();
                q.pop();
                temp.push_back(popped->val);

                if(popped->left)  q.push(popped->left);
                if(popped->right) q.push(popped->right);
            }

            res.push_back(temp);
            temp.clear();
        }
        return res;
    }
};

File: Binary-Tree-BFS/binary-tree-level-order-traversal.py
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []

        q = deque([root]) 
        res = []

        while q:
            temp = []
            for _ in range(len(q)):
                popped = q.popleft()
                temp.append(popped.val)
                
                if popped.left:
                    q.append(popped.left)
                if popped.right:
                    q.append(popped.right)

            res.append(temp)
            temp = []

        return res
